### PR TITLE
Cross platform, new features, documentation, rendering fixes

### PR DESCRIPTION
**Pull request by TyomaVader**
### Added
- CMake support
- Dismiss button for notifications (optional)
- Optional button in the notification that executes a user-defined function
- GitHub Actions for Linux and Windows builds
- Linux support
- Documentation

### Fixed
- Notifications now render above all other windows
- Notifications now render in the correct position when the main window is moved
- Compilation warnings about incorrect usage of ```ImGui::Text()```

### Changed
- Code readability improved
- Switched to Font Awesome 6 icons
- Visual changes to the notifications (can be customized in the ```main.cpp``` file)
- Default Dear ImGui theme changed to *Embrace The Darkness* by [janekb04](https://github.com/janekb04)
- Switched from classic enums to scoped enums
- Upgraded Dear ImGui version used in example to [v1.89.9 WIP](https://github.com/ocornut/imgui/commit/11613013860d149667302a258041dcd832069f36)

### CODE REVIEW

**Review by patrickcjk — APPROVED** [review]

File: unixExample/backends/ImGuiNotify.hpp [hunk 1]
```diff
@@ -0,0 +1,648 @@
...
+		NOTIFY_FORMAT(this->setContent, format);
+
+		this->onButtonPress = onButtonPress;
+		this->setButtonLabel(buttonLabel);
+	}
+};
+
+namespace ImGui
+{
+	inline std::vector<ImGuiToast> notifications;
```
Comment (sirlis) [inline unixExample/backends/ImGuiNotify.hpp:463]: this line require `stdc++17`, which should be explicitly told in README

File: unixExample/backends/ImGuiNotify.hpp [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+#include <vector>			// Vector for storing notifications list
+#include <string>
+#include <chrono>			// For the notifications timed dissmiss
+#include <functional>		// For storing the code, which executest on the button click in the notification
+
+#include "imgui.h"
+#include "imgui_internal.h"
+
+#include "IconsFontAwesome6.h"
+#include "fa_solid_900.h"
```
Comment (sirlis) [inline unixExample/backends/ImGuiNotify.hpp]: windows cmake  fail with this header `fa_solid_900.h`, suggest removing it. No use according to me.

**Comment by patrickcjk** [comment]
Hey! thanks for your contribution, can you please fix the conflict so I can validate merge ?

**Comment by TyomaVader** [comment]
> Hey! thanks for your contribution, can you please fix the conflict so I can validate merge ?

Hi, I've changed the project structure quite a lot, so `example` folder needs to be created again for automatic merge to work. But there are `Unix` and `Windows` examples now, so it can confuse people. Could you please resolve the merge conflict manually instead ? I'm always open to any other solution, if you have one. Thanks!

**Pull request closed without merging**